### PR TITLE
[Impeller] Fix text scaling issues again, this time with perspective when a save layer is involved

### PR DESCRIPTION
**Pull request by dnfield**
Alternative to https://github.com/flutter/engine/pull/43662

Records the basis vector of the current CTM into text contents and reuses that rather than trying to get it again at render time. That method breaks if perspective is involved in the CTM and a subpass gets translated, which can modify the scale (and rotation) of the matrix.

We're definitely not doing things quite right with perspective here, but the real fix to that is to either record the fully transformed glyph into the atlas or to use SDF/path based rendering.

Fixes https://github.com/flutter/flutter/issues/130476

I still have some concerns about how `EntityPass::Render` is a mix of mutations and constness but we can try to address that independently. 

I expect this to re-improve the regression noted in https://github.com/flutter/flutter/issues/130594

### CODE REVIEW

File: impeller/entity/contents/text_contents.h [hunk 1]
```diff
@@ -45,20 +45,26 @@ class TextContents final : public Contents {
...
+      Scalar scale) override;
+
   // |Contents|
   bool Render(const ContentContext& renderer,
               const Entity& entity,
               RenderPass& pass) const override;
 
  private:
   TextFrame frame_;
+  Scalar scale_;
```
Comment (bdero) [inline impeller/entity/contents/text_contents.h:60]: Primitive default:
```suggestion
  Scalar scale_ = 1.0;
```

File: impeller/aiks/aiks_unittests.cc [hunk 2]
```diff
@@ -2877,6 +2877,35 @@ TEST_P(AiksTest, CanCanvasDrawPicture) {
   ASSERT_TRUE(OpenPlaygroundHere(canvas.EndRecordingAsPicture()));
 }
 
+TEST_P(AiksTest, DrawPictureWithText) {
```
Comment (bdero) [inline impeller/aiks/aiks_unittests.cc:2880]: Sorta nit: The text seems to either be invisible or hard to see in the goldens with the golden service's white background. Maybe throw in black backgrounds with `DrawPaint`?

Comment (bdero) [reply impeller/aiks/aiks_unittests.cc:2880]: Actually, seems like the new goldens aren't rendering anything for some reason.

Comment (dnfield) [reply impeller/aiks/aiks_unittests.cc:2880]: Did the gold test part have a chance to run?

The text is red, I thought that should render on either black or white background fine :\

Comment (dnfield) [reply impeller/aiks/aiks_unittests.cc:2880]: But before this change, the new tests would've had validation errors (the atlas would not contain the scale and the test would crash)

**Comment by dnfield** [comment]
I'm really unclear about why the goldens show nothing.

**Review by bdero — APPROVED** [review]
If they're only broken on Skia gold, maybe land now and add a bug for investigating later?

**Comment by dnfield** [comment]
Verified the golden harness produces an image locally, filed an issue to track for later.

**Pull request merged**